### PR TITLE
test: validator_switch_key_quick fix

### PR DESCRIPTION
**Pull request by Longarithm**
Make `validator_switch_key_quick` less flaky. 
The biggest issue is that validators have equal stake, and when `test0` goes offline, there is a chance `test1` gets bigger rewards and higher stake for the next epoch. This causes validators to swap places, and because sticky assignment doesn't work when number of validators is < number of shards, state sync is triggered.

Unfortunately this is not the end of story because sometimes `Missing main transition state proof for block ... and shard 0` can be triggered. And this is legit, because state proof is saved only when node is a validator, case of shadow validation is not considered. We need to fix this later.

Also fixing variable names with correct ones (other & old validator) and reenable garbage_collection test which was fixed a week ago.

### CODE REVIEW

File: pytest/tests/sanity/validator_switch_key_quick.py [hunk 1]
```diff
@@ -31,26 +31,51 @@ def test_validator_switch_key_quick(self):
...
 
+        default_stake = 50000000000000000000000000000000
+        stake_delta = 5000000000000000000000000000000
+
         # Key will be moved from old_validator to new_validator,
         # while the other_validator remains untouched.
         [
-            other_validator,
             old_validator,
+            other_validator,
```
Comment (staffik) [inline pytest/tests/sanity/validator_switch_key_quick.py:41]: sorry for mixing it up :sweat_smile:

File: pytest/tests/sanity/validator_switch_key_quick.py [hunk 2]
```diff
@@ -31,26 +31,51 @@ def test_validator_switch_key_quick(self):
...
+                ["validators", 1, "amount",
+                 str(default_stake - stake_delta)],
+                [
+                    "records", 0, "Account", "account", "locked",
+                    str(default_stake + stake_delta)
+                ],
+                [
+                    "records", 2, "Account", "account", "locked",
+                    str(default_stake - stake_delta)
+                ]
```
Comment (wacban) [inline pytest/tests/sanity/validator_switch_key_quick.py:67]: nit: consider moving to a helper variable or method

**Review by wacban — APPROVED** [review]
🎉

File: pytest/tests/sanity/validator_switch_key_quick.py (hunk 2 above)
Comment (Longarithm) [reply pytest/tests/sanity/validator_switch_key_quick.py:67]: That's why I created #12273 :)

**Pull request merged**